fe/fe-core/src/main/java/com/starrocks/clone/TabletScheduler.java

```diff
@@ -1176,8 +1176,14 @@ private boolean deleteLocationMismatchReplica(TabletSchedCtx tabletCtx, boolean
 
         Set<Pair<String, String>> matchedLocations = new HashSet<>();
         Replica dupReplica = null;
-        //1. delete the unmatched replica
-        for (Replica replica : tabletCtx.getReplicas()) {
+        // 1. delete the unmatched replica
+        // To ensure the correctness of the cleanup process after a rebalance, we iterate the replicas in reverse order.
+        // After cloning a replica from the source BE to a newly added destination BE, the replica on the new BE will
+        // always be at the end of the list. By traversing the list in reverse, we process older replicas first
+        // and preserve the newly added replica, preventing it from being mistakenly deleted.
+        List<Replica> replicas = tabletCtx.getReplicas();
```

Comment:
The larger the index, the newer the replica should be when it joins, right? Why is it older instead?

Reply:
Hi @gengjun-git 
This replica set is managed with a list, where each new addition is appended to the end.
For example, the original List<Replica> is:
- be0 (rack1)
- be1 (rack2)
- be2 (rack3)

After adding a new backend be10 (rack1) and performing rebalance, the List<Replica> becomes:

be0, be1, be2, be10

With the original traversal order, the result is:

- rack1: be0
- rack2: be1
- rack3: be2

Then, when reaching be10 (rack1), it detects that rack1 already exists, so the newly migrated replica on be10 gets removed.

Reply:
OK, I get you, But I think the comments mislead me:
`By traversing the list in reverse, we process older replicas first and preserve the newly added replica, preventing it from being mistakenly deleted.`
should be:
`By traversing the list in reverse, we process newer replicas first and preserve the newly added replica, preventing it from being mistakenly deleted.`

older => newer

Reply:
HI @gengjun-git ，I've modified the comment. Please help review again.